scene/gui/line_edit.cpp

```diff
@@ -268,7 +268,9 @@ void LineEdit::gui_input(const Ref<InputEvent> &p_event) {
 				return;
 			}
 
-			shift_selection_check_pre(b->is_shift_pressed());
+			if (b->is_shift_pressed()) {
+				shift_selection_check_pre(true);
+			}
```

Comment:
Moving the control inside shift_selection_check_pre might not cause unwanted side effects?

Reply:
Shouldn't do as this is the only call that is used for the mouse, though that does raise a good point, we should probably prevent any another input whilst dragging text.

Reply:
Yes, preventing other input while dragging will avoid unwanted behavior, but I believe this needs to be done in another pull request because it should be done in a generic way, not limited to just Lineedit.